src/engine/GroupBy.cpp

```diff
@@ -625,8 +616,8 @@ std::optional<GroupBy::OptimizedGroupByData> GroupBy::checkIfJoinWithFullScan(
 
   // Determine if any of the two children of the join operation is a
   // triple with three variables that fulfills the condition.
-  auto* child1 = static_cast<const Operation*>(join)->getChildren().at(0);
-  auto* child2 = static_cast<const Operation*>(join)->getChildren().at(1);
+  auto* child1 = join.getChildren().at(0);
+  auto* child2 = join.getChildren().at(1);
```

Comment:
It was in purpose that this is a `Join` (we statically already know here that we have the correct type).
The casting back to the base clase here is an implementation detail of the (a little fishy) way, that `getChildren()`
is implemented.
So please leave this as it was, or maybe add a `TODO<robintf> implement `Operation::getChildren` in a way that doesn't require this explicit casting to the base class.
(The clean way would be to have a private ` getChildrenImpl` in all the child classes).